### PR TITLE
Documentation: Write inline documentation for serializer_registry.py

### PR DESCRIPTION
**Pull request by deepeshaburse**
## Description
Contains inline documentation for all 3 functions within serializer_registry.py.

## Relevant Issue
Issue #191 

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/NimbleEdge/EnvisEdge/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/NimbleEdge/EnvisEdge/blob/main/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [NimbleEdge Styleguide](https://numpydoc.readthedocs.io/en/latest/format.html)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/NimbleEdge/EnvisEdge/labels)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by deepeshaburse** [comment]
@ramesht007 Kindly review this PR.

**Comment by ramesht007** [comment]
please follow these documentation guidelines:
pep8 [guidelines](https://www.python.org/dev/peps/pep-0008/)
numpy style [guide](https://numpydoc.readthedocs.io/en/latest/format.html)

and refactor your comments.
thanks

**Comment by deepeshaburse** [comment]
> please follow these documentation guidelines: pep8 [guidelines](https://www.python.org/dev/peps/pep-0008/) numpy style [guide](https://numpydoc.readthedocs.io/en/latest/format.html)
> 
> and refactor your comments. thanks

I have made the required changes, kindly check them.

File: fedrec/serialization/serializer_registry.py [hunk 1]
```diff
@@ -6,12 +6,62 @@
...
+  Arguments
+  ----------
+  serialized_obj_name : str
+  
+  --------------------------------------------------------
+  Returns
+  ---------
+  The value corresponding to that class name in the memory : str
+  
+  --------------------------------------------------------
```
Comment (divyanshugit) [inline fedrec/serialization/serializer_registry.py:24]:
```suggestion
 
```

File: fedrec/serialization/serializer_registry.py [hunk 2]
```diff
@@ -6,12 +6,62 @@
...
     # given object name.
     return Registrable.lookup_class_ref(serialized_obj_name)
 
 
 def serialize_attribute(obj):
+    """
+    This function is used to serialize an attribute
+    over the obj parameter.
+    
+    -------------------------------------------------------
```
Comment (divyanshugit) [inline fedrec/serialization/serializer_registry.py:41]:
```suggestion
    
```

File: fedrec/serialization/serializer_registry.py [hunk 3]
```diff
@@ -32,6 +82,39 @@ def serialize_attribute(obj):
 
 
 def deserialize_attribute(obj: Dict):
+    """
+    Arguments
+    ---------- 
+    obj : dict
+   
+    -----------------------------------------------------
```
Comment (divyanshugit) [inline fedrec/serialization/serializer_registry.py:90]:
```suggestion
    
```

File: fedrec/serialization/serializer_registry.py [hunk 4]
```diff
@@ -32,6 +82,39 @@ def serialize_attribute(obj):
...
+        keys of obj.
+      value 
+        deserialize_attribute() is called recursively on each value in obj.
+      If obj is a dictionary.
+      
+    list or tuple
+      if obj is list or tuple, deserialize_attribute() is called recursively 
+      on each attribute in obj.
+      
+    -----------------------------------------------------
```
Comment (divyanshugit) [inline fedrec/serialization/serializer_registry.py:111]:
```suggestion
    
```

File: fedrec/serialization/serializer_registry.py [hunk 5]
```diff
@@ -32,6 +82,39 @@ def serialize_attribute(obj):
 
 
 def deserialize_attribute(obj: Dict):
+    """
```
Comment (divyanshugit) [inline fedrec/serialization/serializer_registry.py:85]: It will be great, if you can explain the function/method over here.

**Review by divyanshugit — CHANGES_REQUESTED** [review]

**Comment by divyanshugit** [comment]
@deepeshaburse Can you please remove unnecessary lines from your inline comments?

File: fedrec/serialization/serializer_registry.py [hunk 6]
```diff
@@ -32,6 +81,39 @@ def serialize_attribute(obj):
...
+    list or tuple
+      if obj is list or tuple, deserialize_attribute() is called recursively 
+      on each attribute in obj.
+      
+    
+    Raises
+    ---------
+    ValueError
+      If none of the conditions are satisfied.
+
```
Comment (divyanshugit) [inline fedrec/serialization/serializer_registry.py:115]: This whole section, It's quite confusing as in I'm not sure what are you trying to convey over here.

File: fedrec/serialization/serializer_registry.py [hunk 7]
```diff
@@ -6,12 +6,61 @@
...
+    
+    obj : str, int, float or bool
+      If obj is string, integer, float or boolean.
+    
+    obj.serialize() 
+      If the object has an attribute named serialize.
+     
+    "Object must be serializable" : str
+      If the object does not have an attribute named serialize.
+
```
Comment (divyanshugit) [inline fedrec/serialization/serializer_registry.py:62]: Again, it's not clear. If you've any issues understanding this script. Please do let us know

**Comment by divyanshugit** [comment]
@deepeshaburse Sorry earlier I was just going through the structure of comments. Please let us know over here or on discord if you need any sort of help.

**Comment by deepeshaburse** [comment]
> @deepeshaburse Sorry earlier I was just going through the structure of comments. Please let us know over here or on discord if you need any sort of help.

Yes, I'll ping you on discord about this. Thanks for the review!

**Pull request closed without merging**